### PR TITLE
Bug memoryleak

### PR DESCRIPTION
**Pull request by cservan**
Resolved a memory leak issue

### CODE REVIEW

**Comment by maudetes** [comment]
Fixing issue #18

File: models_config.yaml [hunk 1]
```diff
@@ -1 +1,6 @@
-language_identification: resources/lid.176.ftz
\ No newline at end of file
+# name of the model:
+#  - file localization
+#  - language
+language_identification:
```
Comment (maudetes) [inline models_config.yaml]: Maybe change for a key:value system instead of a list ?

File: src/text-classifier.cpp [hunk 2]
```diff
@@ -71,16 +61,17 @@ void ProcessArgs(int argc, char **argv) {
...
-  cout << "Using " << threads << " threads" << endl;
-  cout << "Using port " << port << endl;
-  cout << "Using config file " << model_config << endl;
+//   Address addr(Ipv4::any(), port);
+  cout << "[INFO]\tUsing config file:\t" << API_config << endl;
+  cout << "[INFO]\tCores available:\t" << hardware_concurrency() << endl;
+  rest_server classification_api(API_config, threads, debug);
 
-  rest_server classification_api(addr, model_config, debug);
+//   rest_server classification_api(addr, API_config, debug);
```
Comment (maudetes) [inline src/text-classifier.cpp]: Remove commented lines

File: src/text-classifier.cpp [hunk 3]
```diff
@@ -7,29 +7,27 @@
...
 
 void usage() {
-  cout << "./text-classifier --model-config <filename> [--port <port>] "
-          "[--threads <nthreads>] [--debug]\n\n"
-          "\t--port (-p)              port to use (default 9009)\n"
-          "\t--threads (-t)           number of threads (default 1)\n"
-          "\t--model-config (-f)      config file in which all models are "
-          "described (needed)\n"
+  cout << "./text-classifier --config <filename> [--debug]\n\n"
+//           "\t--port (-p)              port to use (default 9009)\n"
```
Comment (maudetes) [inline src/text-classifier.cpp]: Remove commented lines

File: src/rest_server.cpp [hunk 4]
```diff
@@ -151,32 +250,38 @@ void rest_server::doClassificationBatchPost(const Rest::Request &request,
...
-  
   if (j.find("batch_data") != j.end()) {
     for (auto& it: j["batch_data"]){
       if (it.find("text") != it.end()) {
         string text = it["text"];
-        string tokenized = l_tok.tokenize_str(text);
-        it.push_back(nlohmann::json::object_t::value_type(
-            string("tokenized"), tokenized));
+        string tokenized;
+        it.push_back(nlohmann::json::object_t::value_type(string("tokenized"), tokenized));
```
Comment (maudetes) [inline src/rest_server.cpp]: remove this line, tokenized hasn't been computed yet

File: models_config.yaml (hunk 1 above)
Comment (cservan) [reply models_config.yaml]: Done

File: src/text-classifier.cpp (hunk 2 above)
Comment (cservan) [reply src/text-classifier.cpp]: Done

File: src/text-classifier.cpp (hunk 3 above)
Comment (cservan) [reply src/text-classifier.cpp]: Done

File: src/rest_server.cpp (hunk 4 above)
Comment (cservan) [reply src/rest_server.cpp]: Done

**Comment by cservan** [comment]
All remarks are done

**Comment by cservan** [comment]
@maudetes and I agreed on the modifications done

**Pull request merged**